cli/internal/core/engine.go

```diff
@@ -97,27 +91,40 @@ func (e *Engine) Execute(visitor Visitor, opts EngineExecutionOptions) []error {
 	})
 }
 
-func (e *Engine) getTaskDefinition(taskName string, taskID string) (*Task, error) {
-	pipeline, err := e.getPipelineFromWorkspace(util.RootPkgName)
+func (e *Engine) getTaskDefinition(pkg string, taskName string, taskID string) (*Task, error) {
+	pipeline, err := e.completeGraph.GetPipelineFromWorkspace(pkg, e.isSinglePackage)
+
+	// An error here means there was no turbo.json in the workspace.
```

Comment:
I think this is where we would see the benefit of splitting "not found" from "some error occurred"